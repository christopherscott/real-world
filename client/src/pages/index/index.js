// @flow

import React from 'react';
import { Query, type QueryRenderProps } from 'react-apollo';
import gql from 'graphql-tag';
import MovieList from '../../components/MovieList';
import { type Films } from '../../generated/types.flow';

const FILMS = gql`
  query Films {
    films {
      id
      title
      year
      runtime
      plot
      poster
      rating
      votes
      directors {
        id
        firstName
        lastName
      }
    }
  }
`;

const IndexPage = () => (
  <Query query={FILMS}>
    {({ loading, error, data: { films } }: QueryRenderProps<Films>) => (
      <MovieList loading={loading} error={error} films={films} />
    )}
  </Query>
);

export default IndexPage;
