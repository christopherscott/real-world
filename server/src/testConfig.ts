import { Prisma } from './generated/prisma';

export const db = new Prisma({ endpoint: 'http://localhost:4477' });

export const context = { db, request: {} };

export const clean = async () => {
  await db.mutation.deleteManyGenres({});
  await db.mutation.deleteManyComments({});
  await db.mutation.deleteManyPersons({});
  await db.mutation.deleteManyFilms({});
};
