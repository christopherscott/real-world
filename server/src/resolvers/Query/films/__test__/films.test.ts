import { map, omit, length } from 'ramda';
import { context, db, clean } from '../../../../testConfig';
import fixtures from './fixtures';
import { films, film } from '../index';

afterEach(async () => {
  await clean();
});

test('should return films ordered by raiting', async () => {
  await Promise.all(
    map(data => db.mutation.createFilm({ data }), fixtures.films)
  );

  const result = await films(null, null, context, null);

  expect(length(result)).toBe(5);

  expect(map(omit(['id']), result)).toMatchSnapshot();
});

test('should return film', async () => {
  const { id } = await db.mutation.createFilm(
    { data: fixtures.films[0] },
    '{ id }'
  );

  const result = await film(null, { id }, context, null);

  expect(omit(['id'], result)).toMatchSnapshot();
});
