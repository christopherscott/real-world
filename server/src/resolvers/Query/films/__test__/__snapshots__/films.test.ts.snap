// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return film 1`] = `
Object {
  "plot": "Two imprisoned men bond over a number of years, finding solace and eventual redemption through acts of common decency.",
  "poster": "http://ia.media-imdb.com/images/M/MV5BODU4MjU4NjIwNl5BMl5BanBnXkFtZTgwMDU2MjEyMDE@._V1_SX300.jpg",
  "rating": 9.3,
  "runtime": "1970-01-01T02:22:00.000Z",
  "title": "The Shawshank Redemption",
  "votes": 1078045,
  "year": "1994-01-01T00:00:00.000Z",
}
`;

exports[`should return films ordered by raiting 1`] = `
Array [
  Object {
    "plot": "Two imprisoned men bond over a number of years, finding solace and eventual redemption through acts of common decency.",
    "poster": "http://ia.media-imdb.com/images/M/MV5BODU4MjU4NjIwNl5BMl5BanBnXkFtZTgwMDU2MjEyMDE@._V1_SX300.jpg",
    "rating": 9.3,
    "runtime": "1970-01-01T02:22:00.000Z",
    "title": "The Shawshank Redemption",
    "votes": 1078045,
    "year": "1994-01-01T00:00:00.000Z",
  },
  Object {
    "plot": "The aging patriarch of an organized crime dynasty transfers control of his clandestine empire to his reluctant son.",
    "poster": "http://ia.media-imdb.com/images/M/MV5BMjEyMjcyNDI4MF5BMl5BanBnXkFtZTcwMDA5Mzg3OA@@._V1_SX300.jpg",
    "rating": 9.2,
    "runtime": "1970-01-01T02:55:00.000Z",
    "title": "The Godfather",
    "votes": 762332,
    "year": "1972-01-01T00:00:00.000Z",
  },
  Object {
    "plot": "The early life and career of Vito Corleone in 1920s New York is portrayed while his son, Michael, expands and tightens his grip on his crime syndicate stretching from Lake Tahoe, Nevada to pre-revolution 1958 Cuba.",
    "poster": "http://ia.media-imdb.com/images/M/MV5BNDc2NTM3MzU1Nl5BMl5BanBnXkFtZTcwMTA5Mzg3OA@@._V1_SX300.jpg",
    "rating": 9,
    "runtime": "1970-01-01T03:20:00.000Z",
    "title": "The Godfather: Part II",
    "votes": 496772,
    "year": "1974-01-01T00:00:00.000Z",
  },
  Object {
    "plot": "A dissenting juror in a murder trial slowly manages to convince the others that the case is not as obviously clear as it seemed in court.",
    "poster": "http://ia.media-imdb.com/images/M/MV5BODQwOTc5MDM2N15BMl5BanBnXkFtZTcwODQxNTEzNA@@._V1_SX300.jpg",
    "rating": 8.9,
    "runtime": "1970-01-01T01:36:00.000Z",
    "title": "12 Angry Men",
    "votes": 266350,
    "year": "1957-01-01T00:00:00.000Z",
  },
  Object {
    "plot": "An insomniac office worker looking for a way to change his life crosses paths with a devil-may-care soap maker and they form an underground fight club that evolves into something much, much more...",
    "poster": "http://ia.media-imdb.com/images/M/MV5BMjIwNTYzMzE1M15BMl5BanBnXkFtZTcwOTE5Mzg3OA@@._V1_SX300.jpg",
    "rating": 8.8,
    "runtime": "1970-01-01T02:19:00.000Z",
    "title": "Fight Club",
    "votes": 819812,
    "year": "1999-01-01T00:00:00.000Z",
  },
]
`;
